Bluetooth: Samples: Add a new sample for Encrypted Advertising (#56101)
zephyrproject-rtos/zephyr: samples/bluetooth/encrypted_advertising/README.rst

.. _bluetooth_encrypted_advertising_sample:

Bluetooth: Encrypted Advertising
################################

Overview
********

> This sample demonstrates the usage of the encrypted advertising feature. This
> include:

Reviewer: ```suggestion
This sample demonstrates the use of the encrypted advertising feature,
such as:
```

>  - the exchange of the session key and the initialization vector using the Key
>    Material characteristic;
>  - the encryption of advertising payloads;
>  - the decryption of those advertising payloads;
>  - and finally the update of the Randomizer field whenever the RPA is changed.

Reviewer: ```suggestion
 - the exchange of the session key and the initialization vector using the Key
   Material characteristic,
 - the encryption of advertising payloads,
 - the decryption of those advertising payloads,
 - and the update of the Randomizer field whenever the RPA is changed.
```

> To use the `bt_ead_encrypt` and `bt_ead_decrypt` functions you need to enable

Reviewer: ```suggestion
To use the `bt_ead_encrypt` and `bt_ead_decrypt` functions you, must enable
```

the Kconfig symbol :kconfig:option:`CONFIG_BT_EAD`.

> This sample uses extended advertising, this is **not** mandatory to use the
> feature. The encrypted adverting feature can be used with legacy advertising as
> well.

Reviewer: ```suggestion
While this sample uses extended advertising, it is **not** mandatory when using
the encrypted adverting feature. The feature can be used with legacy advertising
as well.
```

> Hardware Setup

Reviewer: Other Bluetooth samples use a Requirements header, and list device requirements using bullet points.

I suggest matching those more closely for this section.

**************

This sample use two applications, two devices need to be setup.

> The first one should be flashed with the central and the second one with the
> peripheral.

Reviewer: This sentence should be moved to the Building and running section, on its own between first two paragraphs there currently.
```suggestion
Flash one device with the central configuration, and another device with the
peripheral configuration.
```
Not 100% sure that "configuration" is the correct word here, feel free to use a more accurate term.

The two devices should automatically connect based on the device name advertised

> by the peripheral.

Reviewer: This sentence should be moved to Building and Running so it is in the same paragraph and before the sentence that starts "Here are the outputs".

See :ref:`bluetooth samples section <bluetooth-samples>` for details.

This sample use two applications, two devices need to be setup.

> Flash on device with the central application, and another device with the

Reviewer: ```suggestion
Flash one device with the central application, and another device with the
```

.. _bluetooth_encrypted_advertising_sample:

Bluetooth: Encrypted Advertising
################################

Overview
********

This sample demonstrates the use of the encrypted advertising feature, such as:

 - the exchange of the session key and the initialization vector using the Key
   Material characteristic,
 - the encryption of advertising payloads,
 - the decryption of those advertising payloads,
 - and the update of the Randomizer field whenever the RPA is changed.

To use the `bt_ead_encrypt` and `bt_ead_decrypt` functions, you must enable
the Kconfig symbol :kconfig:option:`CONFIG_BT_EAD`.

While this sample uses extended advertising, it is **not** mandatory when using
the encrypted advertising. The feature can be used with legacy advertising as
well.

Requirements
************

* Two boards with Bluetooth Low Energy support
* Two boards with a push button connected via a GPIO pin, see :ref:`Button
  sample <button-sample>` for more details

Building and Running
********************

This sample can be found under
:zephyr_file:`samples/bluetooth/encrypted_advertising` in the Zephyr tree.

See :ref:`bluetooth samples section <bluetooth-samples>` for details.

> This sample use two applications, so two devices need to be setup.

Reviewer 2: uses / requires
